src/client.rs

```diff
@@ -254,37 +387,191 @@ impl DatadogTracing {
             let _ = log::set_boxed_logger(Box::new(tracer.clone()));
             log::set_max_level(lc.level.to_level_filter());
         }
+        if config.enable_tracing {
+            tracing::subscriber::set_global_default(tracer.clone()).unwrap();
+        }
         tracer
     }
 
-    pub fn start_span(&self, trace_id: u64, span_id: u64) -> Result<(), ()> {
+    fn send_log(&self, record: LogRecord) -> Result<(), ()> {
```

Comment:
Do these need to have a `Result` return type? Could you just write
```
fn send_log(&self, record: LogRecord) {
    .lock()
    .unwrap()
    .send(TraceCommand::Log(record));
}
```

Reply:
This would warn because "send" returns a result, so we'd at least have to 
```
send(...).unwrap_or(());
``` 

to clear the warnings.

I just felt it was nice for the caller of send_log et. al. to decide how to handle a channel closure (that's why this might return Err), even though the code currently using it just ignores the error anyway...